addon/utils/locale.js

```diff
@@ -121,6 +121,16 @@ function getFlattenedTranslations(id, owner) {
     assign(result, getFlattenedTranslations(parentId, owner));
   }
 
+  let envConfig = owner.factoryFor('config:environment').class;
+  let ENV = (envConfig.i18n || {});
+  let defaultLocale = ENV.defaultLocale;
+  let defaultFallback = ENV.defaultFallback;
+  if (defaultFallback && defaultLocale) {
```

Comment:
I don't know how expensive `Object.assign` is on objects with a few hundred to a few thousand keys. If it's expensive, we could also check `defaultLocale !== id`.